wandb/sdk/launch/launch.py

```diff
@@ -25,13 +24,27 @@
 _logger = logging.getLogger(__name__)
 
 
-def resolve_agent_config(
+def resolve_agent_config(  # noqa: C901
```

Comment:
lol we should really refactor this

Reply:
I'm calling that out of band for this PR but also you're right